splinterd/src/config/mod.rs

```diff
@@ -36,3 +38,285 @@ pub use crate::config::toml::TomlConfig;
+    }
+
+    pub fn network_endpoint(&self) -> String {
+        self.network_endpoint.0.clone()
+    }
+
+    pub fn peers(&self) -> Vec<String> {
+        self.peers.0.clone()
+    }
+
+    pub fn node_id(&self) -> String {
+        self.node_id.0.clone()
+    }
+
+    pub fn bind(&self) -> String {
+        self.bind.0.clone()
+    }
+
+    #[cfg(feature = "database")]
+    pub fn database(&self) -> String {
+        self.database.0.clone()
+    }
+
+    pub fn registry_backend(&self) -> String {
+        self.registry_backend.0.clone()
+    }
+
+    pub fn registry_file(&self) -> String {
+        self.registry_file.0.clone()
+    }
+
+    pub fn heartbeat_interval(&self) -> u64 {
+        self.heartbeat_interval.0
+    }
+
+    pub fn admin_service_coordinator_timeout(&self) -> Duration {
+        self.admin_service_coordinator_timeout.0
+    }
+
+    pub fn state_dir(&self) -> String {
+        self.state_dir.0.clone()
+    }
+
+    fn storage_source(&self) -> ConfigSource {
+        self.storage.1.clone()
+    }
+
+    fn transport_source(&self) -> ConfigSource {
+        self.transport.1.clone()
+    }
+
+    fn cert_dir_source(&self) -> ConfigSource {
+        self.cert_dir.1.clone()
+    }
+
+    fn ca_certs_source(&self) -> ConfigSource {
+        self.ca_certs.1.clone()
+    }
+
+    fn client_cert_source(&self) -> ConfigSource {
+        self.client_cert.1.clone()
+    }
+
+    fn client_key_source(&self) -> ConfigSource {
+        self.client_key.1.clone()
+    }
+
+    fn server_cert_source(&self) -> ConfigSource {
+        self.server_cert.1.clone()
+    }
+
+    fn server_key_source(&self) -> ConfigSource {
+        self.server_key.1.clone()
+    }
+
+    fn service_endpoint_source(&self) -> ConfigSource {
+        self.service_endpoint.1.clone()
+    }
+
+    fn network_endpoint_source(&self) -> ConfigSource {
+        self.network_endpoint.1.clone()
+    }
+
+    fn peers_source(&self) -> ConfigSource {
+        self.peers.1.clone()
+    }
+
+    fn node_id_source(&self) -> ConfigSource {
+        self.node_id.1.clone()
+    }
+
+    fn bind_source(&self) -> ConfigSource {
+        self.bind.1.clone()
+    }
+
+    #[cfg(feature = "database")]
+    fn database_source(&self) -> ConfigSource {
+        self.database.1.clone()
+    }
+
+    fn registry_backend_source(&self) -> ConfigSource {
+        self.registry_backend.1.clone()
+    }
+
+    fn registry_file_source(&self) -> ConfigSource {
+        self.registry_file.1.clone()
+    }
+
+    fn heartbeat_interval_source(&self) -> ConfigSource {
+        self.heartbeat_interval.1.clone()
+    }
+
+    fn admin_service_coordinator_timeout_source(&self) -> ConfigSource {
+        self.admin_service_coordinator_timeout.1.clone()
+    }
+
+    fn state_dir_source(&self) -> ConfigSource {
+        self.state_dir.1.clone()
+    }
+
+    /// Displays the configuration value along with where the value was sourced from.
+    pub fn log_as_debug(&self) {
```

Comment:
Is there a reason this was chosen over implementing the `Debug` trait and just calling `debug!("{}", config}`? Seems more idiomatic and flexible.

Reply:
I believe that having the separate debug statements will allow for better formatting than just calling debug on the entire object, which is ultimately the intention.

Reply:
I feel like debug!("{}", config) would be better reserved for a single-line piece of information about the config rather than this very long dump of all the configuration.

Reply:
But further - we are looking for more than one line in the debug output. So what is here now LGTM.